git_hg_sync/events.py

```diff
@@ -1,27 +1,31 @@
-from dataclasses import dataclass
+from typing import Self
 
+from pydantic import (
+    BaseModel,
+    Field,
+    model_validator,
+)
 
-@dataclass
-class Push:
+
+class Push(BaseModel):
     repo_url: str
-    branches: dict[
-        str, str
-    ]  # Mapping between branch names (key) and corresponding commit sha (value)
-    tags: dict[
-        str, str
-    ]  # Mapping between tag names (key) and corresponding commit sha (value)
+    branches: dict[str, str] | None = Field(
+        default_factory=dict
+    )  # Mapping between branch names (key) and corresponding commit sha (value)
+    tags: dict[str, str] | None = Field(
+        default_factory=dict
+    )  # Mapping between tag names (key) and corresponding commit sha (value)
     time: int
-    pushid: int
+    push_id: int
     user: str
     push_json_url: str
 
+    @model_validator(mode="after")
+    def check_branch_tags(self) -> Self:
+        """Check that at least one of branches or tags is not empty."""
+        if not self.branches and not self.tags:
+            raise ValueError("Either (non-empty) branches or tags is required")
+        return self
 
-@dataclass
-class Tag:
-    repo_url: str
-    tag: str
-    commit: str
-    time: int
-    pushid: int
-    user: str
-    push_json_url: str
+
+Event = Push
```

Comment:
Any reason not rename `Push` to `Event` in the class definition?

Reply:
I'd like to retain the ability to handle other events, even if we are down to 1 at the moment.